### PR TITLE
Remove applications that depend on python2Packages

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change
A first batch of application removals as well as python3 ports. Looked for application that depend on python2Packages.

https://github.com/NixOS/nixpkgs/issues/148779

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by veprbl** [comment]
Courtesy pings:

- sickrage: @sterfield
- couchpotato: @fadenb
- euca2ools: @edolstra
- zabbix-cli: @womfoo
- tor-arm: @thoughtpolice
- pdf-redact-tools: @leenaars
- polysh: @astsmtl
- pyrit: @danielfullmer
- enpass: @ewok
- creddump: @fishi0x01
- knockknock: @copumpkin
- dtrx: @spwhitt
- blastem: @puffnfresh
- electrum-dash: @np
- pybitmessage: @jgillich
- wifite2: @Lassulus @danielfullmer

**Comment by danielfullmer** [comment]
No objection from me for removing `pyrit`. However,  I don't believe we need to mark `wifite2` as broken, since `pyrit` is only an optional dependency for `wifite2`.

**Comment by mweinelt** [comment]
> No objection from me for removing `pyrit`. However, I don't believe we need to mark `wifite2` as broken, since `pyrit` is only an optional dependency for `wifite2`.

There is a failing test without it.

```
======================================================================
ERROR: testAnalyze (test_Handshake.TestHandshake)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/source/tests/test_Handshake.py", line 27, in testAnalyze
    hs.analyze()
  File "/build/source/wifite/model/handshake.py", line 127, in analyze
    self.divine_bssid_and_essid()
  File "/build/source/wifite/model/handshake.py", line 38, in divine_bssid_and_essid
    pairs = self.pyrit_handshakes()  # Find bssid/essid pairs that have handshakes in Pyrit
  File "/build/source/wifite/model/handshake.py", line 109, in pyrit_handshakes
    return Pyrit.bssid_essid_with_handshakes(
  File "/build/source/wifite/tools/pyrit.py", line 28, in bssid_essid_with_handshakes
    pyrit = Process(command, devnull=False)
  File "/build/source/wifite/util/process.py", line 93, in __init__
    self.pid = Popen(command, stdout=sout, stderr=serr, stdin=stdin, cwd=cwd, bufsize=bufsize)
  File "/nix/store/rppr9s436950i1dlzknbmz40m2xqqnxc-python3-3.9.9/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/nix/store/rppr9s436950i1dlzknbmz40m2xqqnxc-python3-3.9.9/lib/python3.9/subprocess.py", line 1821, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'pyrit'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/build/source/tests/test_Handshake.py", line 29, in testAnalyze
    exit()
  File "/nix/store/rppr9s436950i1dlzknbmz40m2xqqnxc-python3-3.9.9/lib/python3.9/_sitebuiltins.py", line 26, in __call__
    raise SystemExit(code)
SystemExit: None
```

It should be skipped because it is decorated with

```
@unittest.skipUnless(Process.exists('pyrit'), 'pyrit is missing')
```

So I replaced that and it still tries to execute the test.

```
diff --git a/pkgs/tools/networking/wifite2/default.nix b/pkgs/tools/networking/wifite2/default.nix
index 44227dcce2a..9463c1c20c9 100644
--- a/pkgs/tools/networking/wifite2/default.nix
+++ b/pkgs/tools/networking/wifite2/default.nix
@@ -24,6 +24,11 @@ python3.pkgs.buildPythonApplication rec {
     })
   ];
 
+  postPatch = ''
+    substituteInPlace tests/test_Handshake.py \
+      --replace "@unittest.skipUnless(Process.exists('pyrit'), 'pyrit is missing')" "@unittest.skip"
+  '';
+
   propagatedBuildInputs = [
     aircrack-ng
     wireshark-cli
```

Feel free to push a fix.

**Comment by danielfullmer** [comment]
@mweinelt I'm having trouble pushing my change to this branch. Either my permissions aren't sufficient or I'm doing it wrong.  But feel free to cherry-pick this commit: https://github.com/danielfullmer/nixpkgs/commit/5a50edf458772affeca8cc9f887989704478cce4

**Review by Mic92 — APPROVED** [review]

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 153151` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>wifite2</li>
    <li>zabbix-cli</li>
  </ul>
</details>

File: nixos/modules/misc/ids.nix [hunk 1]
```diff
@@ -296,7 +296,7 @@ in
       infinoted = 264;
       sickbeard = 265;
       headphones = 266;
-      couchpotato = 267;
+      # couchpotato = 267; # unused, remove 2022-01-01
```
Comment (dotlambda) [inline nixos/modules/misc/ids.nix:299]:
```suggestion
      # couchpotato = 267; # unused, removed 2022-01-01
```

File: nixos/modules/misc/ids.nix [hunk 2]
```diff
@@ -603,7 +603,7 @@ in
...
```
Comment (dotlambda) [inline nixos/modules/misc/ids.nix:606]:
```suggestion
      # couchpotato = 267; # unused, removed 2022-01-01
```

**Comment by thoughtpolice** [comment]
Yes, look like `tor-arm` has reached the end of its road. Thanks for the heads up!

**Review by jonringer — APPROVED** [review]
diff LGTM

**Comment by Synthetica9** [comment]
Should probably mention this in the release notes for 22.05

**Pull request merged**